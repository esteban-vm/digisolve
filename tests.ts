import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
