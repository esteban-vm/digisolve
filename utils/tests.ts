import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
