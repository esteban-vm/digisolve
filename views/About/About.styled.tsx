import { styled } from '@/styles'

export const Box = styled.div`
  margin-top: 150px;

  & h2 {
    color: #000;
    font-size: 250%;
    font-weight: 500;
    margin-bottom: 50px;
  }

  & p {
    color: #555;
    margin-bottom: 50px;
  }

  & img {
    width: 80%;
    height: auto;
  }

  &:first-child {
    text-align: left;
  }

  &:last-child {
    text-align: right;
  }
`
