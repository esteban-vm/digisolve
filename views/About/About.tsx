import type { Component } from '@/types'
import { useId } from 'react'
import Image from 'next/image'
import { Button, Grid } from '@/components'
import { styled, mediaQuery } from '@/styles'

const AboutComponent: Component = (props) => {
  const id = useId()

  return (
    <section aria-labelledby={id} {...props}>
      <Grid.Row>
        <Grid.Col isHalf>
          <h2 id={id}>A digital agency focused on growing your online presence</h2>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Corporis totam molestiae quod vitae deserunt.
            Architecto fugiat veritatis cum necessitatibus sapiente!
          </p>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Corporis totam molestiae quod vitae deserunt.
            Architecto fugiat veritatis cum necessitatibus sapiente!
          </p>
          <Button text='Read more' />
        </Grid.Col>
        <Grid.Col isHalf>
          <Image src='/img/about-img.jpg' alt='Digisolve app on Laptop' width={1_920} height={1_280} />
        </Grid.Col>
      </Grid.Row>
    </section>
  )
}

const About = styled(AboutComponent)`
  margin-top: 150px;
  padding-bottom: 20px;

  ${mediaQuery('md')} {
    margin-top: 60px;
  }

  h2 {
    color: var(--color-black);
    text-transform: capitalize;
    font-size: 250%;
    font-weight: 500;
    margin-bottom: 50px;

    ${mediaQuery('md')} {
      margin-bottom: 35px;
    }

    ${mediaQuery('sm')} {
      margin-bottom: 20px;
    }
  }

  p {
    color: var(--color-grey);
    margin-bottom: 50px;

    :first-of-type {
      ${mediaQuery('lg')} {
        margin-bottom: 20px;
      }
    }

    :last-of-type {
      ${mediaQuery('sm')} {
        margin-bottom: 30px;
      }
    }
  }

  img {
    width: 80%;
    height: auto;
  }

  .col {
    :first-of-type {
      text-align: left;

      ${mediaQuery('md')} {
        padding-left: 1%;
      }
    }

    :last-of-type {
      text-align: right;

      ${mediaQuery('md')} {
        margin-left: 0;
        text-align: center;
      }
    }
  }
`

export default About
