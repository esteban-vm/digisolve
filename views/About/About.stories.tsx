import type { Meta } from '@storybook/react'
import { styled } from '@/utils/styles'
import About from './About'

const Wrapper = styled.div`
  width: 100%;
  background-color: var(--color-white);
`

export default {
  component: About,
  title: 'Views/About',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
} as Meta<typeof About>

export const AboutExample = () => <About />
