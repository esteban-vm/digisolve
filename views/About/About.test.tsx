import { render, cleanup, screen, create } from '@/tests'
import About from './About'

describe('<About/> test cases:', () => {
  describe('should display:', () => {
    beforeEach(() => {
      render(<About />)
    })

    afterEach(cleanup)

    it('the heading', () => {
      const heading = screen.getByRole('heading')
      expect(heading).toBeInTheDocument()
      expect(heading).toBeVisible()
      expect(heading).toHaveTextContent(/^a digital agency focused on Growing your online presence$/i)
    })

    it('the paragraphs', () => {
      const paragraphs = screen.getAllByRole('paragraph')
      expect(paragraphs).toHaveLength(2)

      for (const paragraph of paragraphs) {
        expect(paragraph).toBeInTheDocument()
        expect(paragraph).toBeVisible()
      }
    })

    it('the button', () => {
      const button = screen.getByRole('button')
      expect(button).toBeInTheDocument()
      expect(button).toBeVisible()
      expect(button).toHaveTextContent(/^read more$/i)
    })

    it('the image', () => {
      const image = screen.getByRole('img')
      expect(image).toBeInTheDocument()
      expect(image).toBeVisible()
      expect(image).toHaveAccessibleName(/^digisolve app on laptop$/i)
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const aboutTree = create(<About />).toJSON()
      expect(aboutTree).toMatchSnapshot()
    })
  })
})
