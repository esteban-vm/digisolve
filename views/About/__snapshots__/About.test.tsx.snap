// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª ABOUT: style tests: should render with correct styles 1`] = `
.emotion-0 {
  margin-top: 150px;
  padding-bottom: 20px;
}

.emotion-0 h2 {
  color: var(--color-black);
  text-transform: capitalize;
  font-size: 250%;
  font-weight: 500;
  margin-bottom: 50px;
}

.emotion-0 p {
  color: var(--color-grey);
  margin-bottom: 50px;
}

.emotion-0 img {
  width: 80%;
  height: auto;
}

.emotion-0 .span_1_of_2:first-of-type {
  text-align: left;
}

.emotion-0 .span_1_of_2:last-of-type {
  text-align: right;
}

.emotion-1 {
  display: inline-block;
}

.emotion-1 a:link,
.emotion-1 a:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  color: var(--color-primary);
}

.emotion-1 a:link:hover,
.emotion-1 a:visited:hover,
.emotion-1 a:link:active,
.emotion-1 a:visited:active {
  background-color: var(--color-primary);
  color: var(--color-white);
}

<section
  aria-labelledby="about_heading"
  className="emotion-0"
>
  <div
    className="row"
  >
    <div
      className="col span_1_of_2"
    >
      <h2
        id="about_heading"
      >
        A digital agency focused on growing your online presence
      </h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Corporis totam molestiae quod vitae deserunt. Architecto fugiat veritatis cum necessitatibus sapiente!
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Corporis totam molestiae quod vitae deserunt. Architecto fugiat veritatis cum necessitatibus sapiente!
      </p>
      <span
        className="emotion-1"
      >
        <a
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          role="button"
        >
          Read more
        </a>
      </span>
    </div>
    <div
      className="col span_1_of_2"
    >
      <img
        alt="Digisolve app on Laptop"
        data-nimg="1"
        decoding="async"
        height={40}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={40}
      />
    </div>
  </div>
</section>
`;
